src/kernels/jupyter/serverSelector.ts

```diff
@@ -212,22 +217,39 @@ export class JupyterServerSelector {
         });
 
         if (uri) {
-            await this.setJupyterURIToRemote(uri);
+            await this.setJupyterURIToRemote(uri, true);
         }
     }
 
     private validateSelectJupyterURI = async (inputText: string): Promise<string | undefined> => {
+        inputText = inputText.trim();
         try {
-            // eslint-disable-next-line @typescript-eslint/no-unused-expressions
             new URL(inputText);
-
-            // Double check http
-            if (!inputText.toLowerCase().includes('http')) {
-                throw new Error('Has to be http');
-            }
         } catch {
             return DataScience.jupyterSelectURIInvalidURI();
         }
+
+        // Double check http
+        if (!inputText.toLowerCase().startsWith('http')) {
+            return 'Has to be http(s)';
+        }
+        // Double check this server can be connected to. Might need a password, might need a allowUnauthorized
+        try {
+            await this.jupyterConnection.validateRemoteUri(inputText);
```

Comment:
We now validate the Uri when its entered, previously its not validated, instead we validate it when we save it into the uri storage and error message is displayed on the bottom right (very disconnected ux)